airflow/www/security.py

```diff
@@ -516,24 +515,25 @@ def _get_all_roles_with_permissions(self) -> Dict[str, Role]:
 
     def create_dag_specific_permissions(self) -> None:
         """
-        Creates 'can_read' and 'can_edit' permissions for all active and paused DAGs.
+        Creates 'can_read' and 'can_edit' permissions for all active and paused DAGs,
+        along with any `access_control` permissions provided in the DAG.
 
         :return: None.
         """
         perms = self.get_all_permissions()
-        rows = (
-            self.get_session.query(models.DagModel.dag_id)
-            .filter(or_(models.DagModel.is_active, models.DagModel.is_paused))
```

Comment:
I believe that `is_active` and `is_paused` are queried for explicitly bc a DAG could be in neither state, in which case we don't want to fetch it to create new permissions.

IIRC it has to do with some historical reason. Something about DAGs getting deleted in the DB; the record sticks around but they've been soft deleted. We'll need to account for that when fetching results.

Reply:
The new approach is the one already taken by the cli and is, from what I can tell, the more complete approach. I believe using `DagBag.collect_dags_from_db` is equivalent to `is_active` (as missing DAGs are removed from `SerializedDagModel`). `is_paused`, not sure why that was explicitly included here but I don't think it needs to be.

Reply:
Yea missing DAGs are removed from `SerializedDagModel` while we mark it as `is_active=False` in `DagModel`

Reply:
I'm going to resolve this, but feel free to chime in again if there are more concerns.